components/de_amp/browser/test/de_amp_util_unittest.cc

```diff
@@ -0,0 +1,153 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/de_amp/browser/de_amp_util.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace de_amp {
+
+/** Test helpers */
+bool CheckIfAmpDetected(const std::string& body, std::string* canonical_link) {
+  return FindCanonicalLinkIfAMP(body, canonical_link);
```

Comment:
nit: missed this earlier, but the convention would be `MaybeFindCanonicalAmpURL`